README.md

```diff
@@ -13,3 +13,15 @@
   complexity suggests a different time.
 
 Add your answers to this markdown file.
+
+1. reason number one that asymptotic analusis could be misleading compared to actual performance is that it doenst always take into acount the size of the data that is being worked with
+2. reason number two is the actual implementation of the algortihm could be very bad even if its fast asymptotically it could be a slow implementation of the algorithm
+3. reason number 3 is the hardware that is being used varies a lot and a very quick asymptotic algorithm could be a lot slower or faster depending on what hardware it is being run on.
+
+I would guess that it takes about 7 seconds in a generic case because binary search is O(log n) type of analysis then it should move up in logrithmic fasion and if there is 1000 to 10000 then the time should move up by about 1.4 times making the time go from 5 seconds to 7 seconds
+
+1. reason number 1 it might take 100 secodns is that the asymptotic complexity is higher than linear and might go up exponentially with larger data sets
+2. reason number 2 is that 10000 might be to big of a data set for the specific hardware or machine making it slow down due to memory constraints
+3. reason number 3 is that there might be a part of the algorithm that is constructed poorly for when handeling larger sizes of data sets that it can run quicker with smaller data sets on.
```

Comment:
For example?

Reply:
Im sorry i am not understanding what you mean by for example in this context.

Reply:
What would be an example of an algorithm that handles a larger dataset poorly?

Reply:
One example would be something like bubble sort that would exponentially get worse as the data set gets larger

Reply:
No, bubble sort doesn't scale exponentially. We're also talking about binary search here, not an arbitrary algorithm.

Reply:
So then are you asking the specifics of how i would need to implement a binary search so that it cant handle a large data set well?

Reply:
No, the question is what specific reasons for binary search to perform this badly could be.

Reply:
would one reason be that the tree could be unbalanced making what should be log n, since it splits it evenly, into just n?

Reply:
If it was linear complexity, what runtime would you expect?

Reply:
if it was linear complexity i would expect the runtime to be the same time as number of elements in the tree

Reply:
Why?

Reply:
Every new element adds a fixed amount of work, and when searching a tree with n elements means that it compares every element until the target is found, so doubling the elements would double the time.

Reply:
Yes, the runtime would double for twice the input size, i.e. you would expect 50 seconds for 10x more -- still less time than was observed. Note that the runtime is *not* the same as the number of elements in the tree.

Reply:
The reason it might be more time still is because asymptotic analysis shows ideal performance but ignores real-world factors like memory use, hardware limits, and inefficient implementation. Binary search can perform worse due to unbalanced trees, cache issues, or overhead so the unexpected runtime (100 seconds vs. 50 seconds) is likely caused by unbalanced trees and hardware constraints not the algorithm itself

Reply:
Okay so its already unbalances so the main difference should be hardware constraints and bottlenecking

Reply:
It could be imbalanced -- the point is that even if it is, that can't cause such a large difference.

Reply:
I re did the readme taking in account everything we talked about and after bouncing some ideas off of other people